test/functional/getblockstats.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#
+# Test getblockstats rpc call
+#
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+)
+
+def assert_contains(data, values, check_cointains=True):
+    for val in values:
+        if (check_cointains):
+            assert(val in data)
+        else:
+            assert(val not in data)
+
+class GetblockstatsTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+        self.extra_args = [['-txindex'], ['-paytxfee=0.003']]
+        self.setup_clean_chain = True
+
+    def run_test(self):
+        node = self.nodes[0]
+        node.generate(101)
+
+        node.sendtoaddress(address=self.nodes[1].getnewaddress(), amount=10, subtractfeefromamount=True)
+        node.generate(1)
+        self.sync_all()
+
+        node.sendtoaddress(address=node.getnewaddress(), amount=10, subtractfeefromamount=True)
+        node.sendtoaddress(address=node.getnewaddress(), amount=10, subtractfeefromamount=True)
+        self.nodes[1].sendtoaddress(address=node.getnewaddress(), amount=1, subtractfeefromamount=True)
+        self.sync_all()
+        node.generate(1)
+
+        start_height = 101
+        max_stat_pos = 2
+        stats = []
```

Comment:
In commit "QA: Test new getblockstats RPC"

Could simplify with list comprehension:

```python
stats = [node.getblockstats(height=start_height + i) for i in range(max_stat_pos+1)]
```